docs/website/docs/plus/features/transformations/dbt-transformations.md

```diff
@@ -0,0 +1,225 @@
+table_reference_adapter(
+    p,
+    "companies",
+    references=[
+        {
+            "referenced_table": "countries",
+            "columns": ["country_id"],
+            "referenced_columns": ["id"],
+        }
+    ],
+)
+
+table_reference_adapter(
+    p,
+    "customers",
+    references=[
+        {
+            "referenced_table": "companies",
+            "columns": ["company_id"],
+            "referenced_columns": ["id"],
+        }
+    ],
+)
+
+table_reference_adapter(
+    p,
+    "orders",
+    references=[
+        {
+            "referenced_table": "customers",
+            "columns": ["customer_id"],
+            "referenced_columns": ["id"],
+        }
+    ],
+)
+
+```
+
+:::note
+Note, that only the relationships that the pipeline is not aware of need to be explicitly passed to the adapter, meaning you don't need to define the parent-child relationships created by `dlt` during the normalization stage, as it will already know about them.
+:::
+
+### 2. Generating your baseline project
+
+Ensure that your dlt pipeline has been run at least once locally or restored from the destination. Then, navigate to the directory where your pipeline is located and, using its name, execute the following command to create a baseline dbt project with dimensional tables for all existing pipeline tables:
+
+```sh
+dlt dbt generate <pipeline-name>
+```
+
+This command generates a new folder named `dbt_<pipeline-name>`, which contains the project with the following structure:
+
+```sh
+dbt_<pipeline-name>/
+├── analysis/
+├── macros/
+├── models/
+│   ├── marts/
+│   │   ├── dim_<pipeline-name>__<table1>.sql
+│   │   ├── dim_<pipeline-name>__<table2>.sql
+│   │   └── dim_<pipeline-name>__<tableN>.sql
+│   ├── staging/
+│   │   ├── sources.yml
+│   │   ├── stg_<pipeline-name>__<table1>.sql
+│   │   ├── stg_<pipeline-name>__<table2>.sql
+│   │   └── stg_<pipeline-name>__<tableN>.sql
+│   ├── <pipeline-name>_dlt_active_load_ids.sql # Used for incremental processing of data
+│   └── <pipeline-name>_dlt_processed_load.sql # Used for incremental processing of data
+├── tests/
+├── dbt_project.yml
+└── requirements.txt
+```
+
+Additionally, in the directory where you ran the generator, you will find a new Python file named `run_<pipeline-name>_dbt.py`, which you can execute to run the project.
+
+
+### 3. Generating fact tables
+
+After creating the base project with dimensional tables, you can create fact tables that will use the previously added relationship hints by running:
+
+```sh
+dlt dbt generate <pipeline-name> --fact <fact_table_name>
+```
+
+The `<fact_table_name>` you provide should be the name of the base table from which the relationships are to be found. This fact table will automatically join all related tables IDs discovered through dlt parent-child relationships, as well as any relationship IDs manually added through the adapter. You can then select and add additional fields in the generated model.
```

Comment:
```suggestion
The `<fact_table_name>` you provide should be the name of the base table in which the relationships are to be found. This fact table will automatically join all related tables IDs discovered through dlt defined parent-child relationships, as well as any relationship IDs manually added through the adapter. You can then select and add additional fields in the generated model.
```

Minor grammatical corrections (I'm not 100% sure I understood this paragrpah correctly though, so let me know if you think they're wrong)